core/src/mindustry/ui/dialogs/JoinDialog.java

```diff
@@ -417,6 +417,9 @@ void refreshCommunity(){
 
             Table[] groupTable = {null, null};
 
+            if(group.prioritized){
+                addHeader(groupTable, group, hidden);
```

Comment:
I haven't tested this, but won't it add an empty, pointless group table if the event server is offline?

Reply:
OK, I tried to fix it, but now there are empty spaces where the unreachable prioritized servers are. Is there a way to make it truly invisible or I have to change the height manually ?

![image](https://github.com/Anuken/Mindustry/assets/62483793/f8027f60-e08e-4456-8964-db71803acd20)

```patch
Index: core/src/mindustry/ui/dialogs/JoinDialog.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/core/src/mindustry/ui/dialogs/JoinDialog.java b/core/src/mindustry/ui/dialogs/JoinDialog.java
--- a/core/src/mindustry/ui/dialogs/JoinDialog.java	(revision 43c7832f0a6b4df6be5408a0ab075097e2ae8eaf)
+++ b/core/src/mindustry/ui/dialogs/JoinDialog.java	(date 1687227381701)
@@ -418,7 +418,7 @@
             Table[] groupTable = {null, null};
 
             if(group.prioritized){
-                addHeader(groupTable, group, hidden);
+                addHeader(groupTable, group, hidden, false);
             }
             //table containing all groups
             for(String address : group.addresses){
@@ -434,9 +434,7 @@
                         || (res.modeName != null && res.modeName.toLowerCase().contains(serverSearch)))) return;
 
                     //add header
-                    if(groupTable[0] == null){
-                        addHeader(groupTable, group, hidden);
-                    }
+                    addHeader(groupTable, group, hidden, true);
 
                     addCommunityHost(res, groupTable[1]);
 
@@ -447,9 +445,15 @@
         }
     }
 
-    void addHeader(Table[] groupTable, ServerGroup group, boolean hidden){ // outlined separately
+    void addHeader(Table[] groupTable, ServerGroup group, boolean hidden, boolean visible){ // outlined separately
+        if(groupTable[0] != null) {
+            groupTable[0].visible = true;
+            return;
+        }
+
         global.table(t -> groupTable[0] = t).fillX().left().row();
 
+        groupTable[0].visible = visible;
         groupTable[0].table(head -> {
             Color col = group.prioritized ? Pal.accent : Color.lightGray;
             if(!group.name.isEmpty()){
```

Reply:
The UI system is pretty awful at dynamically sizing things based on visibility, so the best option here is probably to use a Collapser without an animation.